homeassistant/components/media_player/plex.py

```diff
@@ -62,10 +101,13 @@ def config_from_file(filename, config=None):
 
 def setup_platform(hass, config, add_devices_callback, discovery_info=None):
     """Setup the Plex platform."""
-    config = config_from_file(hass.config.path(PLEX_CONFIG_FILE))
-    if len(config):
+    # get config from plex.conf
+    file_config = config_from_file(hass.config.path(PLEX_CONFIG_FILE))
+    config.update(file_config)  # merge configs
```

Comment:
If an object needs access to both, pass poth objects in. We should not merge the two data sources.